sycamore/llms/openai.py

```diff
@@ -0,0 +1,227 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Optional, Union
+
+from guidance.models import Model
+from guidance.models import OpenAI as GuidanceOpenAI
+from guidance.models import AzureOpenAI as GuidanceAzureOpenAI
+from openai import OpenAI as OpenAIClient
+from openai import AzureOpenAI as AzureOpenAIClient
+from openai import max_retries as DEFAULT_MAX_RETRIES
+from openai.lib.azure import AzureADTokenProvider
+from openai.types.chat import ChatCompletionMessageParam
+from sycamore.llms.llms import LLM
+from sycamore.llms.prompts import GuidancePrompt
+
+
+class OpenAIClientType(Enum):
+    OPENAI = 0
+    AZURE = 1
+
+
+@dataclass
+class OpenAIModel:
+    name: str
+    is_chat: bool = False
+
+
+class OpenAIModels(Enum):
+    TEXT_DAVINCI = OpenAIModel(name="text-davinci-003", is_chat=True)
```

Comment:
this is deprecated now I think?

Reply:
It is, but we added fallback logic, and I didn't want to break that.